components/content_settings/core/browser/brave_content_settings_pref_provider.h

```diff
@@ -20,13 +24,14 @@ namespace content_settings {
 // Because of this reasion, shields configuration was also ephemeral.
 // However, we want shilelds configuration persisted. To do this, we make
 // EphemeralProvider ignore shields type and this class handles.
-class BravePrefProvider : public PrefProvider {
+class BravePrefProvider : public PrefProvider,
```

Comment:
So with cookies and shields, this is actually not a `Pref` provider, since we do not actually store stuff in prefs, but not sure what is the better place for it.

Reply:
Maybe at least it's worth to expand comments regarding all the tricks

Reply:
not sure what you mean, it is a pref provider and that is where the content settings are stored. This is how all persistent content settings are retrieved